### PR TITLE
Further cleanup on getRelatedMemberships - just get them with the api

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Code readability improvement

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------
Per work by @mattwire  it makes sense to just retrieve as an api rather than
do weird object wrangling. This is intended to not change the logic about what is fetched

Comments
----------------------------------------

We do have test coverage of this - eg testIPNPaymentMembershipRecurSuccessNoLeakageOnlineThenOffline

### CODE REVIEW

**Comment by mattwire** [comment]
test this please

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -2476,6 +2480,7 @@ public function loadRelatedObjects(&$input, &$ids, $loadAll = FALSE) {
       }
     }
 
+    // These are probably no longer accessed from anywhere
```
Comment (mattwire) [inline CRM/Contribute/BAO/Contribution.php]: Add @fixme/@todo tag

Comment (eileenmcnaughton) [reply CRM/Contribute/BAO/Contribution.php]: @mattwire you are right - I added this & a few more. Just thinking about it - in the past sometimes I've flagged a bit of a roadmap and even year's later people have followed it so I think I should be more aggressive with my 'todos'

**Comment by mattwire** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
CRM_Core_Payment_AuthorizeNetIPNTest::testIPNPaymentMembershipRecurSuccessNoLeakageOnlineThenOffline
Build timed out (after 60 minutes). Marking the build as aborted.


I can't see how this could cause a loop

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test fail was correct - now fixed

**Comment by mattwire** [comment]
This has good test coverage and tests are now passing.  It is part of an ongoing refactor and doesn't make any functional change in itself.  Code is well commented.  Ok to merge.

**Comment by eileenmcnaughton** [comment]
@mattwire Yay - where to from here - should we extract the innid of the foreach($membership) to a function (updateMembership)  or just move onto trying to clean up the PRs that make actual changes?

**Pull request merged**